proto/redpanda/api/dataplane/v1alpha1/kafka_connect.proto

```diff
@@ -0,0 +1,288 @@
+  RUNNING = 1;
+  //The connector/task has been administratively paused.
+  PAUSED = 2;
+  //The connector/task has failed (usually by raising an exception, which is reported in the status output).
+  FAILED = 3;
+  //The connector/task is restarting.
+  // DESTROYED: Connector is in destroyed state, regardless of any tasks.
+  RESTARTING = 4;
+  //The connector/task has been destroyed.
+  DESTROYED = 5;
+  //The connector/task has been stopped.
+  STOPPED = 6;
+  // The connector/task has not yet been assigned to a worker
+  // UNASSIGNED: Connector is in unassigned state.
+  //				Or Connector is in running state, and there are unassigned tasks.
+  UNASSIGNED = 7;
+  // HEALTHY: Connector is in running state, > 0 tasks, all of them in running state.
+  HEALTHY = 8;
+  // UNHEALTHY: Connector is failed state.
+  //			Or Connector is in running state but has 0 tasks.
+  // 			Or Connector is in running state, has > 0 tasks, and all tasks are in failed state.
+  UNHEALTHY = 9;
+  // DEGRADED: Connector is in running state, has > 0 tasks, but has at least one state in failed state, but not all tasks are failed.
+  DEGRADED = 10;
+}
+
+// buf:lint:ignore ENUM_VALUE_PREFIX
+enum ExpandOption {
+  // buf:lint:ignore ENUM_ZERO_VALUE_SUFFIX
+  UNSPECIFIED = 0;
+  // buf:lint:ignore ENUM_VALUE_UPPER_SNAKE_CASE
+  status = 1;
+  // buf:lint:ignore ENUM_VALUE_UPPER_SNAKE_CASE
+  info = 2;
+}
+
+message ConnectorStatus {
+  message Connector {
+    ConnectorState state = 1;
+    string worker_id = 2;
+  }
+  string name = 1;
+  Connector connector = 2;
+  repeated TaskStatus tasks = 3;
+  string type = 4;
+  string trace = 5;
+}
+
+message TaskStatus {
+  int32 id = 1;
+  ConnectorState state = 2;
+  string worker_id = 3;
+  string trace = 4;
+}
+
+message TaskInfo {
+  string connector = 1;
+  int32 task = 2;
+}
+
+//ConnectorError is the error of a connector, this is holistic error
+//abstraction, made parsing the error trace of connector or Task
+message ConnectorError {
+  // buf:lint:ignore ENUM_VALUE_PREFIX
+  enum Type {
+    // buf:lint:ignore ENUM_ZERO_VALUE_SUFFIX
+    UNSPECIFIED = 0;
+    ERROR = 1;
+    WARNING = 2;
+  }
+  Type type = 1;
+  string title = 2;
+  string content = 3;
+}
+
+//ConectorInfo is the spec of the connector, as defined in the kafka connect
+//api, it can be used as input of the connector creation or output of the
+//conenctors
+message ConnectorSpec {
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+
+  map<string, string> config = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  repeated TaskInfo tasks = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+  string type = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+
+message ListConnectorsRequest {
+  repeated ExpandOption expand = 1 [(buf.validate.field).repeated.max_items = 2];
+  string cluster_name = 2 [
+    (google.api.field_behavior) = REQUIRED,
```

Comment:
Is this field behavior annotation required if we already set the validate annotation? As far as I remember it the buf validate annotations are already considered for client code generation, but I may be wrong. Same applies to the remaining fields where both is set